### PR TITLE
ASM related fixes

### LINKED ISSUE #9011: Can't input EU in GC Machine
### Your GTNH Discord Username

yukieiji

### Your Pack Version

2.1.2.0pre3

### Your Server

Single

### Type of Server

Single Player

### Your Expectation

I tried to input the EU to the GC machine.

### The Reality

Can't inputs the EU(I put the battery directly in and it didn't input.).

### Your Proposal

Fix it.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

### PR DESCRIPTION
**Pull request by Glease**
Replace the now nonexistent soft dependency handling with FML's

For the changes in dependencies.gradle:
1. I have to pull in buildcraft because FML's soft dependency system require the dependency to be on classpath
2. Player stuff is nowhere to be found except on cursemaven (thankfully). I don't know who was using these weird artifact identifiers (really? Use MC version in the version field and API version in the classifier field?) before, but on my system gradle just cannot locate it in any repository specified.
3. Mekanism API for 1710 is nowhere to be found (except in source form) either way. Cursemaven here you go.

Fix https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/9011

### CODE REVIEW

**Comment by glowredman** [comment]
The dependencies were pulled from https://minecraft.curseforge.com/api/maven/ (not to be confused with https://cursemaven.com). Now that you converted all of them, you might aswell remove [this block](https://github.com/GTNewHorizons/Galacticraft/blob/patch-soft-dependency/repositories.gradle#L17-L24).

**Comment by Glease** [comment]
Ah that'd explain it. curseforge maven is notoriously hard to use... It almost always give me a cloudflare captcha page instead of actual file content.

**Comment by glowredman** [comment]
Yeah, I was never able to use it either, I always just copied these three if I had to use it in GC/GS

**Review by mitchej123 — APPROVED** [review]
Should test on Zeta before merging, but looks reasonable.

File: src/main/java/micdoodle8/mods/galacticraft/core/asm/GCTransformer.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static micdoodle8.mods.galacticraft.core.asm.GCLoadingPlugin.debugOutputDir;
+import static micdoodle8.mods.galacticraft.core.asm.GCLoadingPlugin.dev;
+import static org.objectweb.asm.Opcodes.ASM5;
+
+public class GCTransformer implements IClassTransformer {
+    static final Logger log = LogManager.getLogger("TC4TweakTransformer");
```
Comment (glowredman) [inline src/main/java/micdoodle8/mods/galacticraft/core/asm/GCTransformer.java]: Is this supposed to be the same logger as TC4Tweaks?

Comment (Glease) [reply src/main/java/micdoodle8/mods/galacticraft/core/asm/GCTransformer.java]: eh not quite.

**Comment by bombcar** [comment]
> 3. Mekanism API for 1710 is nowhere to be found (except in source form) either way. Cursemaven here you go.

I think curse.maven is good to go but if we care I built Mekanism and so it's in jitpack: 

`compileOnly "com.github.bombcar:Mekanism:1.7.10-SNAPSHOT:api"`

**Pull request merged**